### PR TITLE
Counter v5

### PR DESCRIPTION
**Pull request by adamdickmeiss**

### CODE REVIEW

File: src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java [hunk 1]
```diff
@@ -299,50 +299,70 @@ static String stringOrNull(RequestParameter requestParameter) {
 
   static int getTotalCount(JsonObject reportItem) {
     int count = 0;
-    JsonArray itemPerformances = reportItem.getJsonArray("itemPerformance");
+    JsonArray itemPerformances = reportItem.getJsonArray(altKey(reportItem,
+        "itemPerformance", "Performance"));
```
Comment (MikeTaylor) [inline src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java:303]: Yikes, what horrible crappy work to have to do.

File: src/test/java/org/folio/eusage/reports/MainVerticleTest.java [hunk 2]
```diff
@@ -59,7 +59,7 @@ static JsonObject getCounterReportMock(UUID id, int cnt) {
     report.put("customer", new JsonArray()
         .add(new JsonObject()
             .put("id", "fake customer id")
-            .put("reportItems", new JsonArray()
+            .put(cnt > 1 ? "reportItems" : "Report_Items", new JsonArray()
```
Comment (MikeTaylor) [inline src/test/java/org/folio/eusage/reports/MainVerticleTest.java:62]: Seriously? The name of the JSON field varies in capitalisation and use of underscores dependent on _how many_ records there are??

**Review by MikeTaylor — APPROVED** [review]
Apart from that weird `cnt > 1` check — which no doubt there is a good reason for — this all looks good to me. I'm so sorry you have to do this work, and I fear more incompatible variant of the format coming down the line!

File: src/test/java/org/folio/eusage/reports/MainVerticleTest.java (hunk 2 above)
Comment (adamdickmeiss) [reply src/test/java/org/folio/eusage/reports/MainVerticleTest.java:62]: This is just mock where I want to return both types in one response...

File: src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java (hunk 1 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java:303]: Yes.

File: src/test/java/org/folio/eusage/reports/MainVerticleTest.java (hunk 2 above)
Comment (MikeTaylor) [reply src/test/java/org/folio/eusage/reports/MainVerticleTest.java:62]: Ah, that's a relief!

**Pull request merged**